plugin/pkg/auth/authenticator/token/oidc/oidc_test.go

```diff
@@ -391,5 +391,6 @@ func TestOIDCAuthentication(t *testing.T) {
 		if !reflect.DeepEqual(tt.userInfo, user) {
 			t.Errorf("#%d: Expecting: %v, but got: %v", i, tt.userInfo, user)
 		}
+		client.Close()
```

Comment:
I'm not familiar with this code, but IIUC we're not waiting for client to actually close here. By closing the channel we just notify it that it should, but I don't see how it would prevent the race.